docs/src/modules/components/HomeFooter.js

```diff
@@ -84,9 +84,9 @@ function HomeFooter(props) {
         </Grid>
       </Typography>
       <Typography className={classes.version}>
-        {`Currently v${process.env.LIB_VERSION}. ${t('released')} `}
+        {`${t('currently')} v${process.env.LIB_VERSION}. ${t('released')} `}
```

Comment:
I think that we can introduce a bit of interpolation.

Reply:
https://github.com/martinandert/react-interpolate-component can help.

Reply:
I have tried:
```diff
--- a/docs/src/modules/components/HomeFooter.js
+++ b/docs/src/modules/components/HomeFooter.js
@@ -84,11 +84,12 @@ function HomeFooter(props) {
         </Grid>
       </Typography>
       <Typography className={classes.version}>
-        {`Currently v${process.env.LIB_VERSION}. ${t('released')} `}
-        <Link color="inherit" href="https://github.com/mui-org/material-ui/blob/next/LICENSE">
-          MIT License
-        </Link>
-        {'.'}
+        {t('footer', {
+          version: process.env.LIB_VERSION,
+          link: <Link color="inherit" href="https://github.com/mui-org/material-ui/blob/next/LICENSE">
+            {t('license')}
+          </Link>
+        })}
       </Typography>
     </footer>
   );
diff --git a/docs/src/modules/redux/optionsReducer.js b/docs/src/modules/redux/optionsReducer.js
index a504f4ac5..b7696fff7 100644
--- a/docs/src/modules/redux/optionsReducer.js
+++ b/docs/src/modules/redux/optionsReducer.js
@@ -6,8 +6,9 @@ import mapTranslations from 'docs/src/modules/utils/mapTranslations';

 const req = require.context('docs/translations', false, /translations.*\.json$/);
 const translations = mapTranslations(req, 'json');
+const interpolationRegex = /{{([^{}]*)}}/g;

-const getT = memoize(userLanguage => key => {
+const getT = memoize(userLanguage => {
   const wordings = translations[userLanguage];

   if (!wordings) {
@@ -15,14 +16,21 @@ const getT = memoize(userLanguage => key => {
     return '…';
   }

-  const translation = wordings[key];
+  return (key, substitutions) => {
+    let translation = wordings[key];

-  if (!translation) {
-    console.error(`Missing translation for ${userLanguage}:${key}.`);
-    return `${userLanguage}:${key}`;
-  }
+    if (!translation) {
+      console.error(`Missing translation for ${userLanguage}:${key}.`);
+      return `${userLanguage}:${key}`;
+    }
+
+    translation = translation.replace(interpolationRegex, (expression, argument) => {
+      const replace = substitutions[argument];
+      return replace;
+    });

-  return translation;
+    return translation;
+  };
 });

 const mapping = {
diff --git a/docs/translations/translations.json b/docs/translations/translations.json
index 31cdd4616..982a0cda6 100644
--- a/docs/translations/translations.json
+++ b/docs/translations/translations.json
@@ -1,5 +1,7 @@
 {
   "editDocsColors": "Edit website colors",
+  "footer": "Currently v{{version}}. Released under the {{link}}.",
+  "license": "MIT License",
   "toggleTheme": "Toggle light/dark theme",
   "toggleRTL": "Toggle right-to-left/left-to-right",
   "github": "GitHub repository",
```

but it's not much better.

Reply:
The main advantage is that translators get to see more of the strings in context: `Currently v{{version}}. Released under the {{link}}.`, but we have added context on CrowdIn, so I don't think the sentence fragments are such an issue now. (And they only need to be translated once.)

Reply:
We will talk about it once translators want to switch the words order.

Reply:
That did cross my mind...

Reply:
Ok, as you see fit.

Reply:
Do you have a commit you can push, or shall I apply the diff?

Reply:
The diff doesn't work.

Reply:
Sorry, miunderstood, thought you were proposing the diff as a follow to the react-interpolate-component comment.